maintainers/maintainer-list.nix

```diff
@@ -39,7 +40,81 @@
   Please keep the list alphabetically sorted.
   See `./scripts/check-maintainer-github-handles.sh` for an example on how to work with this data.
   */
-{
+let
+  inherit (lib) types;
+
+  maintainerModule = { config, ... }: {
+    options = {
+      name = lib.mkOption {
+        description = "Your, preferably real, name.";
+        type = types.str;
+      };
+      email = lib.mkOption {
+        description = "Email address.";
+        type = types.str;
+      };
+      github = lib.mkOption {
+        description = ''
+          GitHub handle, as it appears in the URL of your profile page,
+          `https://github.com/<userhandle>`
+        '';
+        type = types.nullOr types.str;
+        default = null;
+      };
+      githubId = lib.mkOption {
+        description = ''
+          GitHub user ID, which can be found at
+          `https://api.github.com/users/<userhandle>`. If `github` is specified
+          but `githubId` isn't, an error is thrown with the correct user ID.
+        '';
+        type = types.nullOr types.ints.unsigned;
+        default = null;
+      };
+      keys = lib.mkOption {
+        description = ''
+          PGP/GPG keys. Add only if you actually use them to sign commits
+          and/or mail.
+
+          To get the required PGP/GPG values for a key run
+          ```shell
+          gpg --keyid-format 0xlong --fingerprint <email> | head -n 2
+          ```
+
+          !!! Note that PGP/GPG values stored here are for informational purposes only, don't use this file as a source of truth.
+        '';
+        type = types.listOf (types.submodule {
+          options.longkeyid = lib.mkOption { type = types.str; };
+          options.fingerprint = lib.mkOption { type = types.str; };
+        });
+        default = [];
+      };
+    };
+  };
+
+  checkMaintainer =
+    let
+      fetchId = github:
+        let response = builtins.fetchurl "https://api.github.com/users/${github}";
```

Comment:
This would break pure evaluation of Nixpkgs.

Reply:
Oh I didn't even see that code. GitHub collapses it because it's too big. Please don't mix code and data.
I recommend to check the ids without Nix. It can be made to work with fixed output derivations; it's a fun exercise, but not something to depend on.

Reply:
@FRidh nixpkgs master should never have this code be evaluating, since it will be checked before that. But yeah I guess I'll remove that, also doesn't play nicely with ofborg

Reply:
It needs to be checked externally anyway I suppose since github's API ratelimit kicks in after 50 requests without key.

Reply:
The code here would only be executed for ones where it's missing, which after this PR should be only single digit numbers ever. But yeah that's another concern.

Reply:
This is now being done as part of a derivation build instead.